Packs/Gem/Integrations/Gem/Gem.py

```diff
@@ -0,0 +1,1211 @@
+
+        Args:
+            action (str): The action to be performed on the entity.
+            entity_id (str): The ID of the entity.
+            entity_type (str): The type of the entity.
+            alert_id (str): The ID of the alert associated with the entity.
+            resource_id (str): The ID of the resource associated with the entity.
+
+        Returns:
+            dict: The response from the API.
+
+        """
+        params = {'action': action, 'entity_id': entity_id, 'entity_type': entity_type,
+                  'alert_id': alert_id, 'resource_id': resource_id}
+
+        response = self.http_request(
+            method='POST',
+            url_suffix=RUN_ACTION_ENDPOINT,
+            params=self._filter_non_empty_params(params)
+        )
+
+        return response
+
+    def add_timeline_event(self, threat_id: str, comment: str, timestamp: str) -> dict:
+        """
+        Adds a timeline event to a threat.
+
+        Args:
+            threat_id (str): The ID of the threat.
+            comment (str): The comment for the timeline event.
+            timestamp (str): The timestamp of the timeline event.
+
+        Returns:
+            dict: The response from the API.
+        """
+        params = {'title': "XSOAR comment", "description": comment, "timeline_event_type": "xsoar", "timestamp": timestamp}
+        response = self.http_request(
+            method='POST',
+            url_suffix=ADD_TIMELINE_EVENT_ENDPOINT.format(id=threat_id),
+            json_data=self._filter_non_empty_params(params)
+        )
+
+        return response
+
+
+''' HELPER FUNCTIONS '''
+# as per recommendation from @freylis, compile once only
+CLEANR = re.compile('<.*?>')
+
+
+def _cleanhtml(raw_html):
+    """
+    Cleans HTML tags from the given raw HTML string.
+
+    Args:
+        raw_html (str): The raw HTML string to be cleaned.
+
+    Returns:
+        str: The cleaned text without HTML tags.
+    """
+    cleantext = re.sub(CLEANR, '', raw_html)
+    return cleantext.replace("\n", "")
+
+
+def _clean_description(alert: dict) -> dict:
+    """
+    Cleans the description of the alert by removing HTML tags.
+
+    Args:
+        alert (dict): The alert dictionary.
+
+    Returns:
+        dict: The modified alert dictionary with cleaned description.
+    """
+    if alert['triage_configuration']['event_groups']:
+        i = 0
+        for e in alert['triage_configuration']['event_groups']:
+            clean_description = _cleanhtml(e['description'])
+            alert['triage_configuration']['event_groups'][i]['description'] = clean_description
+            i += 1
+    return alert
+
+
+def init_client(params: dict) -> GemClient:
+    """
+    Initializes a new GemClient object
+    """
+    return GemClient(
+        base_url=params['api_endpoint'],
+        verify=True,
+        proxy=params.get('proxy', False),
+        client_id=params['client_id'],
+        client_secret=params['client_secret']
```

Comment:
After changing to type 9 since type 4 is deprecated:
```suggestion
        client_id=demisto.getParam('credentials')['identifier'],
        client_secret=demisto.getParam('credentials')['password']
```